components/layout_2020/flow/inline.rs

```diff
@@ -77,6 +81,119 @@ struct LinesBoxes {
+                                    self.current_line.min_content += length;
+                                    self.current_line.max_content += length;
+                                }};
+                            }
+                            macro_rules! add_lengthpercentage {
+                                ($x: expr) => {{
+                                    add_length!($x.length_component());
+                                    self.current_line_percentages += $x.percentage_component();
+                                }};
+                            }
+                            macro_rules! add {
+                                ($condition: ident, $side: ident) => {
+                                    if inline_box.$condition {
+                                        add_lengthpercentage!(padding.$side);
+                                        add_length!(border.$side);
+                                        margin.$side.non_auto().map(|x| add_lengthpercentage!(x));
+                                    }
+                                };
+                            }
+
+                            add!(first_fragment, inline_start);
+                            self.traverse(layout_context, &inline_box.children);
+                            add!(last_fragment, inline_end);
+                        },
+                        InlineLevelBox::TextRun(text_run) => {
+                            let (_, _, _, runs, break_at_start) =
+                                text_run.break_and_shape(layout_context);
+                            if break_at_start {
+                                self.line_break_opportunity()
+                            }
+                            for run in &runs {
+                                let advance = Length::from(run.glyph_store.total_advance());
+                                if run.glyph_store.is_whitespace() {
+                                    self.line_break_opportunity()
+                                } else {
+                                    self.current_line.min_content += advance
+                                }
+                                self.current_line.max_content += advance
+                            }
+                        },
+                        InlineLevelBox::Atomic(atomic) => {
+                            let inner = || {
+                                // atomic
+                                // .intrinsic_inline_sizes
+                                // .as_ref()
+                                // .expect("Accessing intrinsic size that was not requested")
+                                // .clone()
+                                todo!()
+                            };
+                            let (outer, pc) = outer_intrinsic_inline_sizes(&atomic.style, &inner);
+                            self.current_line.min_content += outer.min_content;
+                            self.current_line.max_content += outer.max_content;
```

Comment:
This can be yet another method on `IntrinsicSizes`.

Reply:
Let’s do that if there would be more than one caller. (I did add the other two methods since they replace macros.)